### PR TITLE
Refactor the parameter splitting in ini lookup to handle more path formats

### LINKED ISSUE #16468: ansible playbook failures if paths include spaces
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
ansible --version
ansible 2.1.0.0
  config file =
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

```
cat ansible.cfg
[defaults]

callback_whitelist = profile_tasks
retry_files_enabled = False
host_key_checking = False
inventory = environments/local
callback_plugins = callbacks

[ssh_connection]
scp_if_ssh = True
```
##### OS / ENVIRONMENT

N/A
##### SUMMARY

ansible playbook failures if paths include spaces
##### STEPS TO REPRODUCE

```
ansible-playbook -i environments/mac initdb.yml 
PLAY [ansible] *****************************************************************

TASK [setup] *******************************************************************
Tuesday 28 June 2016  10:47:49 +0200 (0:00:00.028)       0:00:00.028 **********
ok: [ansible]

TASK [include] *****************************************************************
Tuesday 28 June 2016  10:47:50 +0200 (0:00:00.497)       0:00:00.525 **********
included: /Users/lha/Development/hoang anh/openwhisk/ansible/tasks/initdb.yml for ansible

TASK [check if the immortal {{ db_auth }} db with {{ db_provider }} exists?] ***
Tuesday 28 June 2016  10:47:50 +0200 (0:00:00.057)       0:00:00.583 **********
fatal: [ansible]: FAILED! => {"failed": true, "msg": "{{ lookup('ini', 'db_password section=db_creds file={{ playbook_dir }}/db_local.ini') }}: need more than 1 value to unpack"}

PLAY RECAP *********************************************************************
ansible                    : ok=2    changed=0    unreachable=0    failed=1

Tuesday 28 June 2016  10:47:50 +0200 (0:00:00.024)       0:00:00.607 **********
===============================================================================
setup ------------------------------------------------------------------- 0.50s
include ----------------------------------------------------------------- 0.06s
check if the immortal {{ db_auth }} db with {{ db_provider }} exists? --- 0.02s
```

The problem arises from statements like these:

```
db_provider: "{{ lookup('ini', 'db_provider section=db_creds file={{ playbook_dir }}/db_local.ini') }}"
```

where file cannot handle a space in the path.

##### EXPECTED RESULTS

It should handle spaces in the path
##### ACTUAL RESULTS

But it doesn't

```

```

### PR DESCRIPTION
**Pull request by jctanner**
##### ISSUE TYPE
- Bugfix Pull Request
##### ANSIBLE VERSION

```
2.2
```
##### SUMMARY

Fixes #16468

### CODE REVIEW

File: lib/ansible/plugins/lookup/ini.py [hunk 1]
```diff
@@ -25,6 +25,29 @@
...
+def _parse_params(term):
+    '''Safely split parameter term to preserve spaces'''
+
+    keys = ['key', 'section', 'file', 're']
+    params = {}
+    for k in keys:
+        params[k] = ''
+
+    thiskey = 'key'
+    for idp,phrase in enumerate(term.split()):
```
Comment (alikins) [inline lib/ansible/plugins/lookup/ini.py:44]: idp?

**Comment by alikins** [comment]
Any reason this couldn't be done with shlex.split()?

replacing the _parse_params(terms) with shlex.split(terms) seems to pass the
unit tests.

**Comment by bcoca** [comment]
IIRC we tried that at one point and ended up with many issues.

**Comment by jctanner** [comment]
Shlex doesn't work for this ...

```
[jtanner@fedmac AP-16470]$ python
Python 2.7.11 (default, Mar 31 2016, 20:46:51) 
[GCC 5.3.1 20151207 (Red Hat 5.3.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import shlex
>>> x = 'keyD section=sectionD file=/path/with space and/equals/cn=com.ansible'                                
>>> print shlex.split(x)
['keyD', 'section=sectionD', 'file=/path/with', 'space', 'and/equals/cn=com.ansible']
```

**Pull request merged**